gobblin-runtime/src/main/java/gobblin/runtime/local/LocalTaskStateTracker.java

```diff
@@ -88,6 +88,10 @@ public void onTaskRunCompletion(Task task) {
       // it has not reached the maximum number of retries
       WorkUnitState.WorkingState state = task.getTaskState().getWorkingState();
       if (state == WorkUnitState.WorkingState.FAILED && task.getRetryCount() < this.maxTaskRetries) {
+        boolean restartFromFirstIterator = task.getTaskState().getPropAsBoolean(ConfigurationKeys.SOURCE_QUERYBASED_RETRY_RESETITERATOR, false);
```

Comment:
Both `Task` and `LocalTaskStateTracker` are generic. They shouldn't be affected by any source specific configuration.

We're making the change to fix the retry semantic. I don't think you need a configuration key.